### PR TITLE
add vit tf doctest with @add_code_sample_docstrings

### PR DESCRIPTION
**Pull request by johko**
# What does this PR do?

Creates doctest for TF version of ViT. It is a cleaner version of PR #16462 which had a messed up history.

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

@ydshieh

### CODE REVIEW

File: src/transformers/models/vit/modeling_tf_vit.py [hunk 1]
```diff
@@ -756,35 +758,6 @@ def call(
         labels: Optional[Union[np.ndarray, tf.Tensor]] = None,
         training: Optional[bool] = False,
     ) -> Union[TFSequenceClassifierOutput, Tuple[tf.Tensor]]:
-        r"""
-        labels (`tf.Tensor` or `np.ndarray` of shape `(batch_size,)`, *optional*):
-            Labels for computing the image classification/regression loss. Indices should be in `[0, ...,
-            config.num_labels - 1]`. If `config.num_labels == 1` a regression loss is computed (Mean-Square loss), If
-            `config.num_labels > 1` a classification loss is computed (Cross-Entropy).
```
Comment (ydshieh) [inline src/transformers/models/vit/modeling_tf_vit.py:763]: This part should be kept :-) I think.

Check, for example, 

https://github.com/huggingface/transformers/blob/febe42b5daf4b416f4613e9d7f68617ee983bb40/src/transformers/models/beit/modeling_beit.py#L867-L872

**Review by ydshieh — APPROVED** [review]
Thanks a lot @johko!

Very clean PR 💯 . One minor thing to fix, and we are ready to merge it.

**Comment by ydshieh** [comment]
LGTM, thank you @johko !

@sgugger Could you have a final look too?

**Review by sgugger — APPROVED** [review]
LGTM, thanks a lot for your PR!

**Pull request merged**